view/statistics-flow-certificates.js

```diff
@@ -0,0 +1,331 @@
+	return getData('/get-flow-data/', query);
+}, {
+	normalizer: function (args) { return JSON.stringify(args[0]); },
+	max: 1000
+});
+
+var serviceToCertLegacyMatch = { all: [] };
+
+var certificates = {};
+
+var getServiceName = function (ServiceType) {
+	return uncapitalize.call(
+		ServiceType.__id__.slice('BusinessProcess'.length)
+	);
+};
+
+db.BusinessProcess.extensions.forEach(function (ServiceType) {
+	ServiceType.prototype.certificates.map.forEach(function (certificate) {
+		if (!certificates[certificate.key]) {
+			certificates[certificate.key] = { label: certificate.label };
+		}
+		if (!serviceToCertLegacyMatch[getServiceName(ServiceType)]) {
+			serviceToCertLegacyMatch[getServiceName(ServiceType)] = [];
+		}
+		serviceToCertLegacyMatch[getServiceName(ServiceType)].push('certificate-' + certificate.key);
+		serviceToCertLegacyMatch.all.push('certificate-' + certificate.key);
+	});
+});
+
+var buildResultRow = function (rowData) {
+	return assign({
+		submitted: 0,
+		pending: 0,
+		pickup: 0,
+		withdrawn: 0,
+		rejected: 0,
+		sentBack: 0
+	}, rowData || {});
+};
+
+var accumulateResultRows = function (rows) {
+	var result = buildResultRow(rows[0]);
+	rows.slice(1).forEach(function (row) {
+		Object.keys(row).forEach(function (propertyName) {
+			result[propertyName] += row[propertyName];
+		});
+	});
+
+	return result;
+};
+
+var filterData = function (data) {
+	var result = {};
+	location.query.get('service').map(function (service) {
+		location.query.get('certificate').map(function (certificate) {
+			Object.keys(data).forEach(function (key) {
+				if (service !== 'all' && certificate) {
+					result[key] = buildResultRow(data[key][service].certificate[certificate]);
+				} else if (service !== 'all') {
+					result[key] = buildResultRow(data[key][service].businessProcess);
+				} else {
+					var rowsToAccumulate = [];
+					Object.keys(data[key]).forEach(function (service) {
+						if (certificate) {
+							if (!data[key][service].certificate[certificate]) return;
+							rowsToAccumulate.push(data[key][service].certificate[certificate]);
+						} else {
+							rowsToAccumulate.push(data[key][service].businessProcess);
+						}
+					});
+					result[key] = accumulateResultRows(rowsToAccumulate);
+				}
+			});
+		});
+	});
+
+	return result;
+};
+
+var calculateDurationByMode = function (duration, mode) {
+	var result;
+	switch (mode) {
+	case 'weekly':
+		result = Math.ceil(duration.days / 7);
+		break;
+	case 'monthly':
+		result = duration.months;
+		break;
+	case 'yearly':
+		result = duration.years;
+		break;
+	default:
+		result = duration.days;
+		break;
+	}
+
+	return result;
+};
+
+exports['statistics-main'] = function () {
+	var queryHandler, data = new ObservableValue({}), pagination = new Pagination('/flow/');
+	queryHandler = setupQueryHandler(getQueryHandlerConf({
+		processingStepsMeta: this.processingStepsMeta
+	}), location, '/flow/');
+	queryHandler.on('query', function (query) {
```

Comment:
We should use dedicated query handler for this filter, reusing other one configured for other filter bars won't work well. The code here ideally should be as simple as:

```javascript
queryHandler.on('query', function (query) {
  // Query handler's job is to provide us with validated and normalized query object.
  // Ideally we don't have to do any further adaptation, but we have small exceptions as:
  ... convert date instances back to date strings ...
  // then we can just provide query to server, so:
  queryServer(serverQuery).done(...);
}